quota/etcd/quotaapi/conversions_test.go

```diff
@@ -0,0 +1,215 @@
+		Name:      apiSequencingConfig.Name,
+		State:     storagepb.Config_ENABLED,
+		MaxTokens: apiSequencingConfig.MaxTokens,
+		ReplenishmentStrategy: &storagepb.Config_SequencingBased{
+			SequencingBased: &storagepb.SequencingBasedStrategy{},
+		},
+	}
+	storageTimeConfig = &storagepb.Config{
+		Name:      apiTimeConfig.Name,
+		State:     storagepb.Config_DISABLED,
+		MaxTokens: apiTimeConfig.MaxTokens,
+		ReplenishmentStrategy: &storagepb.Config_TimeBased{
+			TimeBased: &storagepb.TimeBasedStrategy{
+				TokensToReplenish:        apiTimeConfig.GetTimeBased().TokensToReplenish,
+				ReplenishIntervalSeconds: apiTimeConfig.GetTimeBased().ReplenishIntervalSeconds,
+			},
+		},
+	}
+)
+
+func TestValidateMask(t *testing.T) {
+	tests := []struct {
+		desc    string
+		mask    *field_mask.FieldMask
+		wantErr bool
+	}{
+		{
+			desc: "commonFields",
+			mask: commonMask,
+		},
+		{
+			desc: "sequencingBased",
+			mask: sequencingBasedMask,
+		},
+		{
+			desc: "timeBased",
+			mask: timeBasedMask,
+		},
+		{
+			desc:    "sequencingAndTime",
+			mask:    &field_mask.FieldMask{Paths: []string{sequencingBasedPath, timeBasedPath}},
+			wantErr: true,
+		},
+		{
+			desc:    "timeAndSequencing",
+			mask:    &field_mask.FieldMask{Paths: []string{timeBasedPath, sequencingBasedPath}},
+			wantErr: true,
+		},
+		{
+			desc:    "unknownPaths",
+			mask:    &field_mask.FieldMask{Paths: []string{statePath, "NOT_A_FIELD", maxTokensPath}},
+			wantErr: true,
+		},
+		{
+			desc:    "namePath", // readonly
+			mask:    &field_mask.FieldMask{Paths: []string{"name"}},
+			wantErr: true,
+		},
+	}
+	for _, test := range tests {
+		err := validateMask(test.mask)
+		if gotErr := err != nil; gotErr != test.wantErr {
+			t.Errorf("%v: validateMask() returned err = %v, wantErr = %v", test.desc, err, test.wantErr)
+		}
+	}
+}
+
+func TestApplyMask(t *testing.T) {
+	srcSequencingConfig := *apiSequencingConfig
+	destSequencingConfig := *storageTimeConfig // mismatched to test overwrite
+	destSequencingConfig.Name = srcSequencingConfig.Name
+
+	srcTimeConfig := *apiTimeConfig
+	destTimeConfig := *storageSequencingConfig // mismatched to test overwrite
+	destTimeConfig.Name = srcTimeConfig.Name
+
+	destClearSequencing := *storageSequencingConfig
+	wantClearSequencing := destClearSequencing
+	wantClearSequencing.ReplenishmentStrategy = nil
+
+	destClearTime := *storageTimeConfig
+	wantClearTime := destClearTime
+	wantClearTime.ReplenishmentStrategy = nil
+
+	tests := []struct {
+		desc       string
+		src        *quotapb.Config
+		dest, want *storagepb.Config
+		mask       *field_mask.FieldMask
+	}{
+		{
+			desc: "applyToBlank",
+			src:  apiTimeConfig,
+			dest: &storagepb.Config{},
+			mask: &field_mask.FieldMask{Paths: []string{statePath, maxTokensPath}},
+			want: &storagepb.Config{
+				State:     storagepb.Config_DISABLED,
+				MaxTokens: apiTimeConfig.MaxTokens,
+			},
+		},
+		{
+			desc: "sequencingBasedOverwrite",
+			src:  &srcSequencingConfig,
```

Comment:
This would be easier to follow if you dropped `srcSequencingConfig` and just used `apiSequencingConfig`, so it's clear the test expects a mapping `apiSequencingConfig` -> `storageSequencingConfig`.